sentinel/src/lib/feed.js

```diff
@@ -84,17 +84,22 @@ const changeQueue = async.queue((change, callback) => {
   });
 });
 
+const resumeProcessing = () => {
+  if (request || changeQueue.paused) {
+    return;
+  }
+  logger.info('transitions: processing enabled');
+  return getTransitionSeq().then(seq => registerFeed(seq));
+};
```

Comment:
It wasn't quite so straightforward, cause now `request` was a promise that resolved into an event emitter and everything about calling it failed. 

The quickest way to fix was to do a double assignment: assign to the promise and then assign to the changes event emitter. We could still end up in a race condition when we cancel (while the request is equal to the promise, it has no `cancel` property to call). 

I think the only clean solution is to have another global variable (`resuming`?) that's true while we're getting the seq. 

What do you think?

Reply:
Hmm yeah the double assign is weird. I'm not too worried about requesting the seq in parallel so perhaps check if request is defined after getting the seq rather than before? eg:

```js
const registerFeed = () => {
  return getTransitionSeq()
    .then(seq => {
      if (request || changeQueue.paused) {
        return;
      }
      logger.info('transitions: processing enabled');
      logger.debug(`transitions: fetching changes feed, starting from ${seq}`);
      request = db.medic.changes({ live: true, since: seq })
         ...
```

Reply:
Ok, I've updated this. 

I put back the `getTransitionSeq()` into the resume function and test for request in register feed.